searchlib/src/vespa/searchlib/tensor/angular_distance.cpp

```diff
@@ -66,25 +66,37 @@ class BoundAngularDistance final : public BoundDistanceFunction {
     }
 };
 
-template class BoundAngularDistance<float>;
-template class BoundAngularDistance<double>;
+template class BoundAngularDistance<float, TemporaryVectorStore<float>>;
+template class BoundAngularDistance<double, TemporaryVectorStore<double>>;
+template class BoundAngularDistance<Int8Float, TemporaryVectorStore<Int8Float>>;
+template class BoundAngularDistance<float, ReferenceVectorStore<float>>;
+template class BoundAngularDistance<double, ReferenceVectorStore<double>>;
+template class BoundAngularDistance<Int8Float, ReferenceVectorStore<Int8Float>>;
 
-template <typename FloatType>
+template <typename FloatType, bool reference_insertion_vector>
 BoundDistanceFunction::UP
-AngularDistanceFunctionFactory<FloatType>::for_query_vector(TypedCells lhs) const {
-    using DFT = BoundAngularDistance<FloatType>;
+AngularDistanceFunctionFactory<FloatType, reference_insertion_vector>::for_query_vector(TypedCells lhs) const {
+    using DFT = BoundAngularDistance<FloatType, TemporaryVectorStore<FloatType>>;
     return std::make_unique<DFT>(lhs);
 }
 
-template <typename FloatType>
+template <typename FloatType, bool reference_insertion_vector>
 BoundDistanceFunction::UP
-AngularDistanceFunctionFactory<FloatType>::for_insertion_vector(TypedCells lhs) const {
-    using DFT = BoundAngularDistance<FloatType>;
-    return std::make_unique<DFT>(lhs);
+AngularDistanceFunctionFactory<FloatType, reference_insertion_vector>::for_insertion_vector(TypedCells lhs) const {
+    if constexpr (reference_insertion_vector) {
```

Comment:
I recommend that you let reference_insertion_vector be a normal argument to the constructor. Then you do not need an extra template argument. Less template arguments and a more local change.

Reply:
Fixed